gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/ApiSubscriptionResource.java

```diff
@@ -201,11 +207,21 @@ public Response changeSubscriptionStatus(
     })
     public Response renewApiKey(
             @PathParam("api") String api,
-            @PathParam("subscription") String subscription) {
-        ApiKeyEntity apiKeyEntity = apiKeyService.renew(subscription);
+            @PathParam("subscription") String subscription,
+            @ApiParam(name = "customApiKey")
+            @CustomApiKey @QueryParam("customApiKey") String customApiKey) {
+
+        if (StringUtils.isNotEmpty(customApiKey)
+                && !parameterService.findAsBoolean(Key.PLAN_SECURITY_APIKEY_CUSTOM_ALLOWED)) {
+            return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
Add information in `@ApiResponses` that the service can return a BAD_REQUEST

Reply:
resolved